pkg/internal/apis/config/validate.go

```diff
@@ -151,7 +152,39 @@ func validatePortMappings(portMappings []PortMapping) error {
 		return fmt.Sprintf("%d/%s", port, protocol)
 	}
 
+	// assignedHostPorts keeps track of assigned host ports
+	assignedHostPorts := make(map[int32]struct{})
+	// getRandomHostPort returns a random port which is not already assigned
+	getRandomHostPort := func() int32 {
+		for {
+			// generate random port in range 5000-65000
+			randomPort := 5000 + rand.Int31n(60000)
+			if _, ok := assignedHostPorts[randomPort]; ok {
+				continue
+			} else {
+				return randomPort
+			}
+		}
+	}
+
 	for _, portMapping := range portMappings {
+		// using default listen address
+		if portMapping.ListenAddress == "" {
+			portMapping.ListenAddress = wildcardAddrIPv4.String()
+		}
+
+		// using random hostport
+		// this is just for validation purpose, actual random host port will be assigned by pkg/cluster/internal/providers
+		if portMapping.HostPort == 0 {
+			portMapping.HostPort = getRandomHostPort()
```

Comment:
I don't think we should fake the random port, it looks like if this value is empty we have to assume there will not be conflict and just skip its validation

Reply:
makes more sense :)